### PR TITLE
docs: add hypervisor reboot guidelines

### PR DESCRIPTION
**Pull request by kpengboy**

### CODE REVIEW

File: ocfweb/docs/docs/staff/procedures/restarting-services.md [hunk 1]
```diff
@@ -19,3 +21,27 @@ Planned restarts of hypervisors should also be announced on this blog, since
 restarting hypervisors can often take several minutes or more.
 
 [status]: http://status.ocf.berkeley.edu
+
+## Rebooting hypervisors
+
+Rebooting hypervisors is a slightly risky business. Hypervisors aren't
+guaranteed to always reboot without problems. Therefore, you shouldn't reboot
```
Comment (jvperrin) [inline ocfweb/docs/docs/staff/procedures/restarting-services.md]: Honestly neither are VMs, but at least it's easier to debug the VMs remotely.

File: ocfweb/docs/docs/staff/procedures/restarting-services.md [hunk 2]
```diff
@@ -19,3 +21,27 @@ Planned restarts of hypervisors should also be announced on this blog, since
...
+them unless you can physically access the lab in case problems arise.
+Additionally, this risk is the reason (D)SM permission is normally required to
+reboot hypervisors.
+
+So you've gotten the necessary permission and made a post on the status blog
```
Comment (jvperrin) [inline ocfweb/docs/docs/staff/procedures/restarting-services.md]: Link to the status blog?

File: ocfweb/docs/docs/staff/procedures/restarting-services.md [hunk 3]
```diff
@@ -19,3 +21,27 @@ Planned restarts of hypervisors should also be announced on this blog, since
...
+(if it's a scheduled restart). What now?
+
+First, gracefully shut down all of the VMs. For planned shutdowns of login
+servers (i.e. tsunami, werewolves and corruption), run the `shutdown` command
+well in advance of the shutdown, so users have impending warning. For other VMs,
```
Comment (jvperrin) [inline ocfweb/docs/docs/staff/procedures/restarting-services.md]: I'd add an example command here, like `sudo shutdown +15 "Rebooting for kernel upgrades"` or something.

File: ocfweb/docs/docs/staff/procedures/restarting-services.md [hunk 4]
```diff
@@ -19,3 +21,27 @@ Planned restarts of hypervisors should also be announced on this blog, since
...
+you can shut them each down via `sudo virsh shutdown`.
+
+Be careful to **always shut down firestorm last**. This is because once
+firestorm is shut down, LDAP logins go offline, and the hypervisors can
+thereafter only be logged into via the root account. It's strongly recommended
+to `sudo -i` before shutting down firestorm.
```
Comment (jvperrin) [inline ocfweb/docs/docs/staff/procedures/restarting-services.md]: Thanks for mentioning this, it's definitely a good point to make

File: ocfweb/docs/docs/staff/procedures/restarting-services.md [hunk 5]
```diff
@@ -19,3 +21,27 @@ Planned restarts of hypervisors should also be announced on this blog, since
...
+
+Once all of the VMs have been shut down, you can then power off the hypervisors
+via `poweroff`.
```
Comment (jvperrin) [inline ocfweb/docs/docs/staff/procedures/restarting-services.md]: Or `shutdown` like mentioned above? Why mix the two?

File: ocfweb/docs/docs/staff/procedures/restarting-services.md (hunk 2 above)
Comment (kpengboy) [reply ocfweb/docs/docs/staff/procedures/restarting-services.md]: See the text above?

Comment (jvperrin) [reply ocfweb/docs/docs/staff/procedures/restarting-services.md]: Sure, I do think making this a link wouldn't hurt though, you can just use the same reference as above, and it helps if someone doesn't read the rest of the sections. I do agree they should read those too, but in my opinion having an extra link is pretty harmless.

File: ocfweb/docs/docs/staff/procedures/restarting-services.md (hunk 4 above)
Comment (chriskuehl) [reply ocfweb/docs/docs/staff/procedures/restarting-services.md]: Even if logins (passwd) are cached, you won't be able to sudo if kerberos is down (auth won't work).

File: ocfweb/docs/docs/staff/procedures/restarting-services.md (hunk 5 above)
Comment (kpengboy) [reply ocfweb/docs/docs/staff/procedures/restarting-services.md]: Eh...well I like `poweroff` because `len("poweroff") < len("shutdown -h now")`

**Pull request merged**